pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,124 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def right_canonicalize_mps(mps):
+    r"""Transform a matrix product state into its right-canonical form.
+
+    Args:
+        mps (list[TensorLike]): List of tensors representing the MPS.
+
+    Returns:
+        A list of tensors representing the MPS in right-canonical form.
+
+    .. seealso:: :class:`~.MPSPrep`.
+
+    A right-canonicalized MPS is a matrix product state where each tensor :math:`A^{(j)}` satisfies
+    the following orthonormality condition:
+
+    .. math::
+
+        \sum_{d_{j,1}, d_{j,2}} A^{(j)}_{d_{j, 0}, d_{j, 1}, d_{j, 2}} \left( A^{(j)}_{d'_{j, 0}, d_{j, 1}, d_{j, 2}} \right)^* = \delta_{d_{j, 0}, d'_{j, 0}}
+
+    **Example**
+
+    .. code-block::
+
+        n_sites = 4
+
+        mps = (
```

Comment:
No need to apply black to the example. Something like this seems more compact to me:

```
mps = ([np.ones((1, 2, 4))] +
       [np.ones((4, 2, 4)) for _ in range(1, n_sites - 1)] +
       [np.ones((4, 2, 1))])
```